src/DataTypes/NestedUtils.cpp

```diff
@@ -114,6 +113,24 @@ Block flatten(const Block & block)
             else
                 res.insert(elem);
         }
+        else if (const DataTypeTuple * type_tuple = typeid_cast<const DataTypeTuple *>(elem.type.get()))
+        {
+            if (type_tuple->haveExplicitNames())
+            {
+                const DataTypes & element_types = type_tuple->getElements();
+                const Strings & names = type_tuple->getElementNames();
+                const ColumnTuple * column_tuple = typeid_cast<const ColumnTuple *>(elem.column.get());
```

Comment:
I think this can be `ColumnConst` also. 🤔 
So similar logic should be added as for the Array above.